dev-lang/perl/perl-5.34.0.recipe

```diff
@@ -0,0 +1,127 @@
+SUMMARY="Larry Wall's Practical Extraction and Report Language"
+DESCRIPTION="Perl was originally developed by Larry Wall in 1987 as a \
+general-purpose Unix scripting language to make report processing easier.
+
+Perl borrows features from other programming languages including C, shell \
+scripting (sh), AWK, and sed. The language provides powerful text processing \
+facilities without the arbitrary data-length limits of many contemporary Unix \
+tools, facilitating easy manipulation of text files.
+
+In addition to CGI, Perl is used for graphics programming, system \
+administration, network programming, finance, bioinformatics, and other \
+applications. Perl is nicknamed 'the Swiss Army chainsaw of scripting \
+languages' because of its flexibility and power, and possibly also because of \
+its perceived 'ugliness'."
+HOMEPAGE="https://www.perl.org/"
+COPYRIGHT="1993-2019 Larry Wall and others"
+LICENSE="GNU GPL v1
+	Artistic"
+REVISION="1"
+perlShortVersion="${portVersion%.*}"
+SOURCE_URI="http://www.cpan.org/src/perl-$portVersion.tar.gz"
+CHECKSUM_SHA256="551efc818b968b05216024fb0b727ef2ad4c100f8cb6b43fab615fa78ae5be9a"
+PATCHES="perl-$portVersion.patchset"
+
+ARCHITECTURES="x86_gcc2 x86 x86_64 ?arm"
+
+PROVIDES="
+	# assume that any perl commands are backwards compatible to version 5.
+	perl = $portVersion compat >= 5
+	cmd:corelist = $portVersion compat >= 5
+	cmd:cpan = $portVersion compat >= 5
+	cmd:enc2xs = $portVersion compat >= 5
+	cmd:encguess = $portVersion compat >= 5
+	cmd:h2ph = $portVersion compat >= 5
+	cmd:h2xs = $portVersion compat >= 5
+	cmd:instmodsh = $portVersion compat >= 5
+	cmd:json_pp = $portVersion compat >= 5
+	cmd:libnetcfg = $portVersion compat >= 5
+	cmd:perl = $portVersion compat >= 5
+	cmd:perl$portVersion = $portVersion compat >= 5
+	cmd:perlbug = $portVersion compat >= 5
+	cmd:perldoc = $portVersion compat >= 5
+	cmd:perlivp = $portVersion compat >= 5
+	cmd:perlthanks = $portVersion compat >= 5
+	cmd:piconv = $portVersion compat >= 5
+	cmd:pl2pm = $portVersion compat >= 5
+	cmd:pod2html = $portVersion compat >= 5
+	cmd:pod2man = $portVersion compat >= 5
+	cmd:pod2text = $portVersion compat >= 5
+	cmd:pod2usage = $portVersion compat >= 5
+	cmd:podchecker = $portVersion compat >= 5
+	cmd:podselect = $portVersion compat >= 5
+	cmd:prove = $portVersion compat >= 5
+	cmd:ptar = $portVersion compat >= 5
+	cmd:ptardiff = $portVersion compat >= 5
+	cmd:ptargrep = $portVersion compat >= 5
+	cmd:shasum = $portVersion compat >= 5
+	cmd:splain = $portVersion compat >= 5
+	cmd:streamzip = $portVersion compat >= 5
+	cmd:xsubpp = $portVersion compat >= 5
+	cmd:zipdetails = $portVersion compat >= 5
+	# vendor_perl refers to the path to packaged perl modules, which includes
+	# the short perl version, so it is backwards compatible up this one.
+	vendor_perl = $perlShortVersion compat >= 5
```

Comment:
I changed this, because I have problems when switching between perl 5.32 and perl 5.34 (conflicting version for other packages using vendor_perl and pkgman was asking to uninstall packages e.g. git)

Reply:
The idea of this dependency is that perl modules are specific to a given Perl version. That's because of paths like "/boot/system/non-packaged/lib/perl5/site_perl/$Config{version}/$Config{archname}" which have the complete version in it.

So, modules built for one version will not work with another.

pkgman is correct when it wants to uninstall things: they will not work out of the box with the new Perl version because this path needs to be changed (and probably the perl modules need to be rebuilt).

We have the same problem for Python, and what we did there is allow multiple packages to be installed (so we have python3, python38 and python39 packages) to allow progressive migration from one to another (after rebuilding all packages).

Reply:
Ok then I was just wrong and will reuse old value for compat

Thank you

Reply:
But then I have this: 

![Screenshot from 2021-06-25 11-12-37](https://user-images.githubusercontent.com/580360/123400687-5149a380-d5a6-11eb-8fb5-b15726fe670e.png)

Is it because the new version of perl is not in official haikuports repo?

Reply:
If you recompile git locally it should use the new Perl version. Then it shuold be possible to install that new Git version.

As I said, we made some changes to Python packages to handle this case (there are multiple packages with different names, and they can be installed side by side). You can look at how it's done there, maybe. Or we can just rebuild everything that depends on Perl and have only the new version remain in the repository (with all perl-dependant packages adjusted to rely on the new version)

Reply:
OK I tested what you said and pushed a new commit to revert this change